include/flatbuffers/flatbuffers.h

```diff
@@ -441,29 +442,29 @@ class Array {
   // operation. For primitive types use @p Mutate directly.
   // @warning Assignments and reads to/from the dereferenced pointer are not
   //  automatically converted to the correct endianness.
-  template<typename U = T>
-  typename std::enable_if<false == flatbuffers::is_scalar<T>::value, U *>::type
+  typename flatbuffers::conditional<scalar_tag::value, void, T *>::type
   GetMutablePointer(uoffset_t i) const {
-    FLATBUFFERS_ASSERT(i < size());
-    return const_cast<T *>(&data()[i]);
-  }
+    #if defined(FLATBUFFERS_ALIGN_CHECK) && !defined(FLATBUFFERS_CPP98_STL)
+    // The data_ is aligned as this Array<T,length>.
+    // The Array<T,length> themself must be aligned not stricker than <T>.
```

Comment:
Some typos in the comments here and below.

Reply:
Thank you, I see.